ios/RCTWKWebView/RCTWKWebView.m

```diff
@@ -232,10 +232,19 @@ - (void)stopLoading
 - (void)setSource:(NSDictionary *)source
 {
   if (![_source isEqualToDictionary:source]) {
+    NSString *customAgent = source[@"customUserAgent"];
+    NSString *oldAgent = _source[@"customUserAgent"];
     _source = [source copy];
     _sendCookies = [source[@"sendCookies"] boolValue];
-    if ([source[@"customUserAgent"] length] != 0 && [_webView respondsToSelector:@selector(setCustomUserAgent:)]) {
+    if ([customAgent length] != 0 && [_webView respondsToSelector:@selector(setCustomUserAgent:)]) {
       [_webView setCustomUserAgent:source[@"customUserAgent"]];
+    } else {
+      [[NSUserDefaults standardUserDefaults] registerDefaults:@{@"UserAgent": customAgent}];
+      [[NSUserDefaults standardUserDefaults] synchronize];
+    }
+    if (![customAgent isEqualToString:oldAgent]) {
+      [self reload];
```

Comment:
it is strange setSource method call "reload" method.

Reply:
Because the library itself will call setSource when set a new user-agent. So I will reload here and simply return to fix the bug rollback when set a new user-agent.

Reply:
I can remove this code. But it make user agent only take effect on new tabs. is it ok?

Reply:
Of cause OK. And you should know that if you want to do that, you should reload upper layer of your code.

Reply:
Yes, I have removed reload in setSource. Please take a look